src/contracts/atlas/AtlasVerification.sol

```diff
@@ -563,7 +537,7 @@ contract AtlasVerification is EIP712, DAppIntegration, AtlasConstants {
         // Verify the signature before storing any data to avoid
         // spoof transactions clogging up dapp userNonces
 
-        bool signatureValid = SignatureChecker.isValidSignatureNow(userOp.from, _hashTypedDataV4(_getUserOpHash(userOp)), userOp.signature);
+        bool signatureValid = SignatureChecker.isValidSignatureNow(userOp.from, _hashTypedDataV4(userOp.getUserOperationHash(UserOperationHashType.DEFAULT)), userOp.signature);
```

Comment:
@ephess in the audit issue you suggest to use the `FULL` hash when calling `isValidSignatureNow`. However here you use `DEFAULT`.

You mentioned including the signature in the hash might be a requirement for eg. Safe smart wallet. Would you have any documentation to point me to for this, as I don't exactly understand how this could be done.

Reply:
> @ephess in the audit issue you suggest to use the FULL hash when calling isValidSignatureNow. However here you use DEFAULT.

Well that's a bit silly. I should fix that.

> You mentioned including the signature in the hash might be a requirement for eg. Safe smart wallet. Would you have any documentation to point me to for this, as I don't exactly understand how this could be done.

I think you'd just use the FULL signature mode? Am I missing something here?

Related, this is the most useful article I found on understanding this EIP, see:

https://ethereum.org/en/developers/tutorials/eip-1271-smart-contract-signatures/

In particular why I think this matters:

> Verification: call isValidSignature. In the message parameter, pass in the message to be verified. In the signature parameter, pass in each safe owner’s individual signatures all concatenated together, back-to-back. The Safe will check that there are enough signatures to meet the threshold and that each signature is valid. If so, it will return a value indicating successful signature verification.

There is another mode in Safe wallets, where you pass in "0x" instead of signatures concatenated together, and it will instead verify the message in a different way.

Reply:
> I think you'd just use the FULL signature mode? Am I missing something here?

Maybe I misunderstood. In FULL signature mode, the `signature` field of the userOp is included in the hashed message to be verified against... that same signature.

As a user, or else, I sign my userOp struct excluding the `signature` field (since I haven't sign it yet). Then update my `userOp.signature` with what I just signed. The validation of that signature must be made against my userOp struct hash excluding that signature field, or it will be invalid.

Safe seems to handling validation in different ways, depending on the `signature` parameter of `isValidSignature`. But I don't see any evidence that they need that signature included in the hashed message?

Beside, this comment might indicates that Atlas won't be compatible with Safe: https://github.com/safe-global/safe-smart-account/blob/main/contracts/handler/CompatibilityFallbackHandler.sol#L58

Reply:
Yeah I also don't think FULL method will work - circular dependency to sign some data which includes the signature to be generated from this signing process.

I think DEFAULT and TRUSTED are all we need, and the smart wallet verification should use DEFAULT

Reply:
Hmm, OK. I think I'm getting myself a little confused here - I was trying to follow the EIP-712 spec for this: https://eips.ethereum.org/EIPS/eip-712#definition-of-hashstruct

In particular:

> hashStruct(s : 𝕊) = keccak256(typeHash ‖ encodeData(s)) where typeHash = keccak256(encodeType(typeOf(s)))

I think I'm getting confused by the cases where the signature field is not actually a signature, which is what we were discussing in #250 Ben. So for example, in the cases where the signature is not actually a signature, but instead some other instruction that's known at signing time, this would work just fine. But in the case when it's an actual signature, then it is in fact a circular dependency.

I will change this to use DEFAULT, and drop FULL.